### PR TITLE
feat: adds global access policy +  adds override vars for existing pols

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Adds `access-global.rego` + attachment to all automated stacks
* Adds `*_policy_id_override` variables for overriding the module's policies

## why
* By default, Spacelift does not provide access for normal users to any stacks. [Access Policies](https://docs.spacelift.io/concepts/policy/stack-access-policy) need to be attached to stacks to grant access. The `access-global.rego` policy provides `write` access to all users by default. 
* The override policy ID vars provide more flexibility in case any module consumers want to customize their stack policies beyond what the module provides by default. 

## references
* N/A

### CODE REVIEW

**Comment by Gowiem** [comment]
@aknysh @osterman getting the below for the terratests on this module. Seems API keys are not being passed to the `provider` block, but that has been happening consistently for a while now. Do we not have a test spacelift account for this module? Anything blocking a review here?

**Comment by aknysh** [comment]
> @aknysh @osterman getting the below for the terratests on this module. Seems API keys are not being passed to the `provider` block, but that has been happening consistently for a while now. Do we not have a test spacelift account for this module? Anything blocking a review here?

We don't require Terratest to pass on this module since we don't have a testing Spacelift account yet

**Comment by Gowiem** [comment]
@aknysh gotcha 👍

File: variables.tf [hunk 1]
```diff
@@ -104,6 +104,30 @@ variable "trigger_global_enabled" {
   default     = false
 }
 
+variable "access_policy_id_override" {
```
Comment (aknysh) [inline variables.tf:107]:
```suggestion
variable "access_policy_id" {
```

@Gowiem thanks, it LGTM.
Maybe we don't need to use `_override`, it should be clear w/o it that we want to provide an existing policy

File: variables.tf [hunk 2]
```diff
@@ -104,6 +104,30 @@ variable "trigger_global_enabled" {
...
+  type        = string
+  description = "ID of an Access policy to override the default"
+  default     = null
+}
+
+variable "push_policy_id_override" {
```
Comment (aknysh) [inline variables.tf:113]:
```suggestion
variable "push_policy_id" {
```

File: variables.tf [hunk 3]
```diff
@@ -104,6 +104,30 @@ variable "trigger_global_enabled" {
...
+  type        = string
+  description = "ID of a Push policy to override the default"
+  default     = null
+}
+
+variable "plan_policy_id_override" {
```
Comment (aknysh) [inline variables.tf:119]:
```suggestion
variable "plan_policy_id" {
```

File: variables.tf [hunk 4]
```diff
@@ -104,6 +104,30 @@ variable "trigger_global_enabled" {
...
+  type        = string
+  description = "ID of a Plan policy to override the default"
+  default     = null
+}
+
+variable "trigger_policy_id_override" {
```
Comment (aknysh) [inline variables.tf:125]:
```suggestion
variable "trigger_policy_id" {
```

File: main.tf [hunk 5]
```diff
@@ -37,15 +37,23 @@ module "stacks" {
   worker_pool_id = var.worker_pool_id
   runner_image   = var.runner_image
 
-  push_policy_id    = spacelift_policy.push.id
-  plan_policy_id    = spacelift_policy.plan.id
-  trigger_policy_id = spacelift_policy.trigger_dependency.id
+  access_policy_id  = var.access_policy_id_override != null ? var.access_policy_id_override : spacelift_policy.access.id
```
Comment (aknysh) [inline main.tf:40]:
```suggestion
  access_policy_id  = var.access_policy_id != null ? var.access_policy_id : spacelift_policy.access.id
```

File: main.tf [hunk 6]
```diff
@@ -37,15 +37,23 @@ module "stacks" {
...
+  push_policy_id    = var.push_policy_id_override != null ? var.push_policy_id_override : spacelift_policy.push.id
```
Comment (aknysh) [inline main.tf:41]:
```suggestion
  push_policy_id    = var.push_policy_id != null ? var.push_policy_id : spacelift_policy.push.id
```

File: main.tf [hunk 7]
```diff
@@ -37,15 +37,23 @@ module "stacks" {
...
+  plan_policy_id    = var.plan_policy_id_override != null ? var.plan_policy_id_override : spacelift_policy.plan.id
```
Comment (aknysh) [inline main.tf:42]:
```suggestion
  plan_policy_id    = var.plan_policy_id != null ? var.plan_policy_id : spacelift_policy.plan.id
```

File: main.tf [hunk 8]
```diff
@@ -37,15 +37,23 @@ module "stacks" {
...
+  trigger_policy_id = var.trigger_policy_id_override != null ? var.trigger_policy_id_override : spacelift_policy.trigger_dependency.id
```
Comment (aknysh) [inline main.tf:43]:
```suggestion
  trigger_policy_id = var.trigger_policy_id != null ? var.trigger_policy_id : spacelift_policy.trigger_dependency.id
```

File: main.tf (hunk 7 above)
Comment (aknysh) [reply main.tf:42]: also update all policies to not create them if we provide IDs of existing policies

```
resource "spacelift_policy" "plan" {  
  count = var.plan_policy_id == null ? 1 : 0
  
  type = "PLAN"  
  name = "Global Plan Policy"  
  body = file("${path.module}/policies/plan-global.rego")
}
```

**Review by aknysh — CHANGES_REQUESTED** [review]
thanks @Gowiem please see comments

**Comment by aknysh** [comment]
@Gowiem  the PR looks good, we need many of the changes :)
I can hijack your PR to improve it and add some other features that we need now, if you don't mind.

**Comment by Gowiem** [comment]
Good stuff @aknysh -- Let me know how I can help. I'm happy to review when you're finished up.

**Comment by Gowiem** [comment]
@aknysh ready to merge this or did you want to hold off for more changes that you wanted to make?

**Review by aknysh — APPROVED** [review]
thanks @Gowiem

**Comment by Gowiem** [comment]
Much appreciated @aknysh!

**Pull request merged**